lib/OpenQA/Script/CloneJob.pm

```diff
@@ -270,7 +270,7 @@ sub clone_job ($jobid, $url_handler, $options, $post_params = {}, $jobs = {}, $d
     $settings->{CLONED_FROM} = $url_handler->{remote_url}->clone->path("/tests/$jobid")->to_string;
     if (my $group_id = $job->{group_id}) { $settings->{_GROUP_ID} = $group_id }
     clone_job_apply_settings($options->{args}, $relation eq 'children' ? 0 : $depth, $settings, $options);
-    OpenQA::Script::CloneJobSUSE::detect_maintenance_update($jobid, $url_handler, $settings);
+    OpenQA::Script::CloneJobSUSE::detect_maintenance_update($jobid, $url_handler, $settings) unless $settings->{PUBLIC_CLOUD_SKIP_MU};
```

Comment:
Since all of this is very SUSE-specific we said it should be part of `OpenQA::Script::CloneJobSUSE`. Now `PUBLIC_CLOUD_SKIP_MU` would "leak" outside that module. So I suggest to add that early return within the function itself.

Reply:
Thanks @Martchus, I've added the verification inside `detect_maintenance_update`, hope it's better this way.